src/sgame/sg_bot_util.cpp

```diff
@@ -2569,7 +2569,17 @@ void BotPain( gentity_t *self, gentity_t *attacker, int )
 		&& attacker->s.eType == entityType_t::ET_PLAYER
 		&& self->botMind->skillSet[BOT_B_PAIN] )
 	{
-
+		if ( G_Team( self ) == TEAM_HUMANS
+			 && self->botMind->goal.targetsValidEntity()
+			 && G_Team( self->botMind->goal.getTargetedEntity() ) == TEAM_ALIENS
+		     && self->botMind->goal.getTargetedEntity()->s.eType != entityType_t::ET_PLAYER
+		     && attacker->s.eType == entityType_t::ET_PLAYER )
+		{
```

Comment:
I suppose the underlying problem here is that `action fight` doesn't do any prioritization of targets once it has latched onto something. Could we check that the running node is `action fight` to make sure it does not randomly abort other actions? (could use the function pointer or the name string you just added)

Reply:
I'm not sure if looking for `action fight` is the right thing to do. I wrote this code long ago, and i can vaguely remember thinking about looking for `action fight` too. But it feels like an unneccessary complication. If a bot has an enemy player as target, it is guaranteed to be fighting it. What else could it do, using our current actions?

Reply:
Technically `action rush` also targets enemy buildables. But I'm not really worried about the standard BTs. This could be annoying if you are trying to script behaviors for a tutorial, a tower defense mode, something like that. If you wanted to program the bots to head straight for an enemy building, it would be impossible because this code keeps deleting the goal.

Reply:
Fair enough. Another method would be to provide more control over how enemies are chosen in the bt files. Including what to do in case of pain.